src/schemas/defineSignatoryInfo.schema.ts

```diff
@@ -0,0 +1,89 @@
+import * as Joi from '@hapi/joi'
+
+import { SignatorySigning } from 'app/models/form/defineSignatoryInfo.model'
+import DirectorToSign from 'app/models/session/directorToSign.model'
+
+const SIGNING_ERROR: string = 'Select how the director will be signing the application'
+const CONTACT_DETAILS_ERROR: string = 'Enter contact details for the directors'
+
+export default function selectSignatoriesSchema(signatories: DirectorToSign[]): Joi.ObjectSchema {
+  return Joi.object(generateSchemaForSignatories(signatories))
+}
+
+function generateSchemaForSignatories(signatories: DirectorToSign[]): Joi.SchemaMap {
+  return signatories.reduce((schema: Joi.SchemaMap, signatory: DirectorToSign) => ({
+    ...schema,
+    ...generateSchemaForSignatory(signatory)
+  }), {})
```

Comment:
Array reduce is a functional way of aggregating an array into a single value. A quick explanation:

Imagine a scenario where 2 signatories will be displayed, with respective IDs `123` and `456`.

Step 1 - Default value of `{}` is assigned.
Step 2 - Append schema fields for first signatory:
```
{
  isSigning_123: Joi.string()...,
  directorEmail_123: Joi.string()...,
  onBehalfName_123: Joi.string()...,
  onBehalfEmail_123: Joi.string()...
}
```
Step 3 - Append schema fields for second signatory:
```
{
  isSigning_123: Joi.string()...,
  directorEmail_123: Joi.string()...,
  onBehalfName_123: Joi.string()...,
  onBehalfEmail_123: Joi.string()...,
  isSigning_456: Joi.string()...,
  directorEmail_456: Joi.string()...,
  onBehalfName_456: Joi.string()...,
  onBehalfEmail_456: Joi.string()...,
}
```
...and so on

Some useful links:
* Array reduce - https://www.freecodecamp.org/news/reduce-f47a7da511a9/
* Merge objects - https://www.javascripttutorial.net/object/javascript-merge-objects/